pkcs11/src/backend/key.rs

```diff
@@ -562,7 +562,7 @@ fn extract_key_id_location_header(headers: HashMap<String, String>) -> Result<St
     let location_header = headers.get("location").ok_or(Error::InvalidData)?;
     let key_id = location_header
         .split('/')
-        .last()
+        .next_back()
```

Comment:
Is there a new clippy lint?

Reply:
Yes: https://rust-lang.github.io/rust-clippy/master/index.html#double_ended_iterator_last

This actually makes me wonder why the `std::str::Split` implementation doesn't optimise this.
And why there is not a lint for iterators that implement `DoubleEndedIterator` but not optimize the call to `last`.

Reply:
Ah, [found it](https://rust-lang.github.io/rust-clippy/master/index.html#double_ended_iterator_last).

Reply:
Even `std::iter::Map` does not optimize this. Kinda disappointed. And I'm not sure it would not be considered "breaking" to change this behaviour because of side-effects in the `Map` call.

Reply:
The [`Iterator::last`](https://doc.rust-lang.org/std/iter/trait.Iterator.html#method.last) documentation describes it as fully evaluating the iterator, so I think it would be incorrect (or at least a breaking change) to implement it more efficiently.

Reply:
The documentation of `last` says it will evaluate all elements, so I guess it can't be optimize for anything with side effects.

Reply:
See also: https://github.com/rust-lang/rust-clippy/issues/14444